File: dmoj/aac5p1/aac5p1.cpp
#include <bits/stdc++.h>
using namespace std;

int n;
int numEven, numOdd;

int main(){
    cin >> n;
    for(int i = 0; i < n; i++){
        int a;
        cin >> a;
        if(a % 2 == 1) numOdd++;
        else numEven++;
    }
    cout << numOdd/2 + numEven/2 << '\n';
}

File: dmoj/aac5p1/aac5p1.py
n = int(input())
arr = list(map(int, input().split()))

num_even = 0
num_odd = 0
for a in arr:
    if a % 2 == 1:
        num_odd += 1
    else:
        num_even += 1

print(num_even//2 + num_odd//2)
